### ISSUE TITLE
*x{IO} bizarre copying

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2036]
Migrated from [rt.perl.org#3314](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3314) (status was 'resolved')

Searchable as RT3314$

**Comment by p5pRT** [comment]
### From @simoncozens



Noted by Arnar Hrafnkelsonn&#8203;:
% perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
Bizarre copy of UNKNOWN in sassign at \-e line 1\.

I'll look into this soon\. But\, man\, that's weird\.

<details><summary>Perl Info</summary>

```


Site configuration information for perl v5.6.0:

Configured by root at Thu Apr  6 14:58:40 JST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.3.99-pre3, archname=i686-linux
    uname='linux bastard 2.3.99-pre3 #6 thu apr 6 03:33:42 jst 2000 i686 unknown '
    config_args='-d'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O2', gccversion=2.95.2 20000313 (Debian GNU/Linux)
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccflags ='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldbm -ldb -ldl -lm -lc -lposix -lcrypt
    libc=/lib/libc-2.1.3.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    v5.6.0-RC1


@INC for perl v5.6.0:
    /usr/local/lib/perl5/5.6.0/i686-linux
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/simon
    LANG=C
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/usr/local/sbin:/usr/sbin:/sbin:/opt/kde/bin
    PERL_BADLANG (unset)
    SHELL=/usr/bin/zsh


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]



Noted by Arnar Hrafnkelsonn&#8203;:
% perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
Bizarre copy of UNKNOWN in sassign at \-e line 1\.

I'll look into this soon\. But\, man\, that's weird\.

<details><summary>Perl Info</summary>

```


Site configuration information for perl v5.6.0:

Configured by root at Thu Apr  6 14:58:40 JST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.3.99-pre3, archname=i686-linux
    uname='linux bastard 2.3.99-pre3 #6 thu apr 6 03:33:42 jst 2000 i686
unknown '
    config_args='-d'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O2', gccversion=2.95.2 20000313 (Debian GNU/Linux)
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccflags ='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE
-D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldbm -ldb -ldl -lm -lc -lposix -lcrypt
    libc=/lib/libc-2.1.3.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    v5.6.0-RC1


@INC for perl v5.6.0:
    /usr/local/lib/perl5/5.6.0/i686-linux
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/simon
    LANG=C
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
   
PATH=/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/usr/local/sbin:/usr/sbin:/sbin:/opt/kde/bin
    PERL_BADLANG (unset)
    SHELL=/usr/bin/zsh


```
</details>

**Comment by p5pRT** [comment]
### From @smpeters

On Thu Jun 01 23&#8203;:43&#8203;:29 2000\, simon@&#8203;deep\-dark\-truthful\-mirror\.perlhacker\.org wrote&#8203;:
> 
> This is a bug report for perl from simon@&#8203;brecon\.co\.uk \(Simon Cozens\)\,
> generated with the help of perlbug 1\.28 running under perl v5\.6\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Noted by Arnar Hrafnkelsonn&#8203;:
> % perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
> Bizarre copy of UNKNOWN in sassign at \-e line 1\.
> 
> I'll look into this soon\. But\, man\, that's weird\.
> 
> 
In Perl 5\.10\, we get a new error message\.

\[steve@&#8203;kirk ~\]$  perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
Not a SCALAR reference at \-e line 1\.

**Comment by p5pRT** [comment]
### From @smpeters

On Mon\, May 19\, 2008 at 10&#8203;:11 PM\, Steve Peters via RT
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> On Thu Jun 01 23&#8203;:43&#8203;:29 2000\, simon@&#8203;deep\-dark\-truthful\-mirror\.perlhacker\.org wrote&#8203;:
>>
>> This is a bug report for perl from simon@&#8203;brecon\.co\.uk \(Simon Cozens\)\,
>> generated with the help of perlbug 1\.28 running under perl v5\.6\.0\.
>>
>>
>> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>> \[Please enter your report here\]
>>
>> Noted by Arnar Hrafnkelsonn&#8203;:
>> % perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
>> Bizarre copy of UNKNOWN in sassign at \-e line 1\.
>>
>> I'll look into this soon\. But\, man\, that's weird\.
>>
>>
> In Perl 5\.10\, we get a new error message\.
>
> \[steve@&#8203;kirk ~\]$  perl \-e 'open\(A\, "/dev/null"\); $bizarre = $\{\*A\{IO\}\}'
> Not a SCALAR reference at \-e line 1\.
>
>

Well\, the changed error in Perl 5\.10 seems to be planned\.  It was
fixed in change \#27179\.

It has been added to Perl 5\.10\, but I would guess that the change is
minor enough for Perl 5\.8\.  Any objections?

Steve Peters
steve@&#8203;fisharerojo\.org

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
